modules/storages/app/workers/storages/health_status_mailer_job.rb

```diff
@@ -40,14 +40,15 @@ class HealthStatusMailerJob < ApplicationJob
     )
 
     def perform(storage:)
-      return unless Storages::Storage.exists?(storage.id)
+      return unless ::Storages::Storage.exists?(storage.id)
```

Comment:
It is not related to your changes. But this line seems to be obsolete to me. Because ActiveJob will already raise an error if there is no storage with such id. It can be just removed(then error will be raised) or rewritten as(no errors):
```ruby 
    def perform(storage_id:)
      storage = Storages::Storage.find_by(id: storage_id)
      return unless storage
```
The error will be:
```
ActiveJob::DeserializationError: Error while trying to deserialize arguments: Couldn't find Storages::NextcloudStorage with 'id'=2 [WHERE \"storages\".\"provider_type\" = $1]
```

cc @apfohl

Reply:
I agree, fyi- it should be a global id as the full storage is passed. Either way, it should fail if it doesn't exist. But perhaps @apfohl can chime in in case we're missing sth.

The top level namespacing was necessary for unit testing, we didn't have these before which is why it wasn't an issue. Running the `health_status_mailer_job_spec` should reproduce it.

Reply:
@ba1ash But then you get an error that needs to be either handled or ignored. Leaving this out would feel a bit incomplete. The purpose to have it in the code is also documentation what happens during `perform`.

I would leave it in for that reason.

Reply:
I do not see how this line helps. If storage is absent in the database then it we never reach it. If storage is present then we do need it. So, I added an explicit `discard_on ActiveJob::DeserializationError`.